### PR TITLE
Fix bitcoind message processing lag

### LINKED ISSUE #82: Fix bitcoind lag
https://github.com/re-vault/revaultd/commit/a4690dfc334a3daaf71885b4e2b60afc358479a6 was purely dumb, but i don't want to get back to 100% CPU..

### PR DESCRIPTION
**Pull request by darosior**
Based on #86 
Fixes #82

### CODE REVIEW

File: src/daemon/bitcoind/actions.rs [hunk 1]
```diff
@@ -587,16 +546,102 @@ pub fn bitcoind_main_loop(
...
+                        BitcoindError::Custom(format!(
+                            "Sending wallet transaction to main thread: {}",
+                            e
+                        ))
+                    })?;
+            }
+        }
     }
+
+    log::info!("Gamse over");
```
Comment (edouardparis) [inline src/daemon/bitcoind/actions.rs:644]: `s/gamse/game`
```suggestion
    log::info!("Exiting");
```

File: src/daemon/database/actions.rs [hunk 2]
```diff
@@ -386,33 +396,50 @@ pub fn db_update_presigned_tx(
...
-                    params![VaultStatus::Secured as u32, vault_id],
-                )
-                .map_err(|e| {
-                    DatabaseError(format!("Updating vault to 'secured': {}", e.to_string()))
-                })?;
+            {
+                // Nope. Mark the vault as 'secured'
+                db_tx
+                    .execute(
+                        "UPDATE vaults SET status = (?1) WHERE id = (?2) ",
```
Comment (edouardparis) [inline src/daemon/database/actions.rs:424]: update `updated_at`

File: src/daemon/database/actions.rs [hunk 3]
```diff
@@ -386,33 +396,50 @@ pub fn db_update_presigned_tx(
...
+                    .map_err(|e| {
+                        DatabaseError(format!("Updating vault to 'secured': {}", e.to_string()))
+                    })?;
+            }
+
+            // Was it the unvault that was fully signed ? If so, mark the vault as active.
+            if is_unvault {
+                db_tx
+                    .execute(
+                        "UPDATE vaults SET status = (?1) WHERE id = (?2) ",
```
Comment (edouardparis) [inline src/daemon/database/actions.rs:436]: update `updated_at`

**Comment by edouardparis** [comment]
oups comments should be for #86

File: src/daemon/bitcoind/actions.rs (hunk 1 above)
Comment (darosior) [reply src/daemon/bitcoind/actions.rs:644]: Lol that's some debugging leftover :sweat_smile:

File: src/daemon/database/actions.rs (hunk 2 above)
Comment (darosior) [reply src/daemon/database/actions.rs:424]: Oh right

File: src/daemon/bitcoind/actions.rs [hunk 4]
```diff
@@ -601,16 +546,98 @@ pub fn bitcoind_main_loop(
...
+        })
+    } else {
+        log::trace!(
+            "Got '{:?}' from bitcoind when requesting wallet transaction '{}'",
+            res,
+            txid
+        );
+        None
+    }
+}
```
Comment (edouardparis) [inline src/daemon/bitcoind/actions.rs]:
```rust
if res.is_err() {
    log::trace!(
            "Got '{:?}' from bitcoind when requesting wallet transaction '{}'",
            res,
            txid
        );
}

res.map_or_else(|(hex, blockheight, received_time)| Some(WalletTransaction{hex, blockheight, received_time}), None)
```

File: src/daemon/bitcoind/actions.rs [hunk 5]
```diff
@@ -601,16 +546,98 @@ pub fn bitcoind_main_loop(
...
+    // Used to shutdown the poller thread
+    let shutdown = Arc::new(AtomicBool::new(false));
+
+    // We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
+    let pol_revaultd = revaultd.clone();
+    let pol_bitcoind = bitcoind.clone();
+    let pol_sp = sync_progress.clone();
+    let pol_sd = shutdown.clone();
+    let poller_thread =
+        std::thread::spawn(move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd));
```
Comment (edouardparis) [inline src/daemon/bitcoind/actions.rs:607]: maybe no make it clear these vars are for the thread
```rust
// We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
    let poller_thread = std::thread::spawn({
        let pol_revaultd = revaultd.clone();
        let pol_bitcoind = bitcoind.clone();
        let pol_sp = sync_progress.clone();
        let pol_sd = shutdown.clone();
        move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd)
    });
```

Comment (darosior) [reply src/daemon/bitcoind/actions.rs:607]: oO i can just use the variables in the closue if i move the `move` ? Thanks!

File: src/daemon/bitcoind/actions.rs (hunk 4 above)
Comment (darosior) [reply src/daemon/bitcoind/actions.rs]: Meh.. This was not changed and i'm not sure it makes it clearer (maybe more clever, but not sure about clearer :p)

Comment (edouardparis) [reply src/daemon/bitcoind/actions.rs]: ok

File: src/daemon/bitcoind/actions.rs (hunk 5 above)
Comment (darosior) [reply src/daemon/bitcoind/actions.rs:607]: This made it much cleaner, applied:
```rust
diff --git a/src/daemon/bitcoind/actions.rs b/src/daemon/bitcoind/actions.rs
index 4e7dffc..9d5055a 100644
--- a/src/daemon/bitcoind/actions.rs
+++ b/src/daemon/bitcoind/actions.rs
@@ -599,12 +599,13 @@ pub fn bitcoind_main_loop(
     let shutdown = Arc::new(AtomicBool::new(false));
 
     // We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
-    let pol_revaultd = revaultd.clone();
-    let pol_bitcoind = bitcoind.clone();
-    let pol_sp = sync_progress.clone();
-    let pol_sd = shutdown.clone();
-    let poller_thread =
-        std::thread::spawn(move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd));
+    let poller_thread = std::thread::spawn({
+        let _revaultd = revaultd.clone();
+        let _bitcoind = bitcoind.clone();
+        let _sync_progress = sync_progress.clone();
+        let _shutdown = shutdown.clone();
+        move || poller_main(_revaultd, _bitcoind, _sync_progress, _shutdown)
+    });
 
     for msg in rx {
         match msg {
```

Will use that from now on!

**Review by edouardparis — APPROVED** [review]
ACK fbda881e83d4d09fb35f31afcdde177e61984722

**Pull request merged**